pkg/cache/cache_test.go

```diff
@@ -0,0 +1,211 @@
+package cache_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-psap/special-resource-operator/pkg/cache"
+	"github.com/openshift-psap/special-resource-operator/pkg/clients"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestCache(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Cache Suite")
+}
+
+func resetGlobals() {
+	clients.Interface = nil
+
+	cache.Node.Count = 0xDEADBEEF
+	cache.Node.List = &unstructured.UnstructuredList{
+		Object: map[string]interface{}{},
+		Items:  []unstructured.Unstructured{},
+	}
+}
+
+var _ = Describe("Cache", func() {
+	Context("Nodes", func() {
+		BeforeEach(resetGlobals)
+		AfterEach(resetGlobals)
+
+		When("machingLabels=nil", func() {
+			When("force=false", func() {
+				When("no taints", func() {
+					It("should work as expected", func() {
+						l := v1.NodeList{
+							Items: []v1.Node{
+								{ObjectMeta: metav1.ObjectMeta{Name: "node1"}},
+								{ObjectMeta: metav1.ObjectMeta{Name: "node2"}},
+							},
+						}
+
+						clients.Interface = &clients.ClientsInterface{
+							Client: fake.NewClientBuilder().WithLists(&l).Build(),
+						}
+
+						err := cache.Nodes(nil, false)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(cache.Node.List.Items).To(HaveLen(len(l.Items)))
+
+					})
+				})
+
+				DescribeTable("taints",
+					func(te v1.TaintEffect) {
```

Comment:
Just for consistency, maybe lets have
```
DescribeTable("taints", func(te v1.TaintEffect) {
```

like in all other places?

Reply:
I'd rather do
```go
DescribeTable(
    "taints",
    func(te v1.TaintEffect) {},
    entries...)
```
WDYT?

Reply:
This will just be different than the indention we have for each one of `Describe`, `Context`, `It`.

Reply:
Isnt this the outcome of `go fmt`?

Reply:
On my setup, `go fmt` does not rewrite this. But I think it's much more consistent to either have all arguments on the same line, or to have each on a new line.